### PR TITLE
kvserver: remove data race in onSpanConfigUpdate

### PR DESCRIPTION
**Pull request by tbg**
We weren't holding `Store.mu`.

Fixes: XXX [repeated 100 times]

Release note: None

### CODE REVIEW

**Comment by tbg** [comment]
@irfansharif did some prep work for when your day begins. Mind taking this over from me? It will close a lot of issues, I think :-)

File: pkg/kv/kvserver/store.go [hunk 1]
```diff
@@ -2309,69 +2309,80 @@ func (s *Store) onSpanConfigUpdate(ctx context.Context, updated roachpb.Span) {
...
-		func(ctx context.Context, it replicaOrPlaceholder) error {
-			repl := it.repl
-			if repl == nil {
-				return nil // placeholder; ignore
-			}
+	// Collect the affected replicas separately, to avoid holding the Store
+	// mutex for longer than necessary.
+	var repls []*Replica
+	{
+		s.mu.Lock()
```
Comment (tbg) [inline pkg/kv/kvserver/store.go:2316]: RLock

**Review by aliher1911 — APPROVED** [review]
lgtm

**Comment by tbg** [comment]
bors r=aliher1911
TFTR!

**Pull request closed without merging**